generator/generator.go

```diff
@@ -342,29 +356,102 @@ func typeSpecs(f *ast.File) []*ast.TypeSpec {
 	return result
 }
 
-func processInterface(fs *token.FileSet, currentPackage *packages.Package, it *ast.InterfaceType, types []*ast.TypeSpec, typesPrefix string, imports []*ast.ImportSpec) (methods methodsList, err error) {
+func getEmbeddedMethods(t ast.Expr, fs *token.FileSet, currentPackage *packages.Package, types []*ast.TypeSpec, pr typePrinter, typesPrefix string, imports []*ast.ImportSpec, params genericsParams) (param genericsParam, methods methodsList, err error) {
```

Comment:
It seems like there are several functions that accept this set of parameters:
fs *token.FileSet
types []*ast.TypeSpec
imports []*ast.ImportSpec
currentPackage *packages.Package

May be it worth creating a structure to pass these things around these methods?

Reply:
No problem, I agree, I was just keeping the same style, the focus was support generics with `*ast.IndexListExpr` and `*ast.IndexList`.

Reply:
@hexdigest please check again